### ISSUE TITLE
connecting to an iLO with self-signed certificate on OSX

### ISSUE THREAD

**Issue by rexroof** [seveas/python-hpilo#76]
This command works in fedora 21:
```
$ hpilo_cli -l Administrator -p secret 10.xxx.xxx.229  get_fw_version
>>> pprint(my_ilo.get_fw_version())
{'firmware_date': 'Nov 07 2014',
 'firmware_version': '2.03',
 'license_type': 'iLO 4 Advanced',
 'management_processor': 'iLO4'}
```

but on OSX 10.10 I get errors:

```
$ hpilo_cli -l Administrator -p secret 10.xxx.xxx.229  get_fw_version
Traceback (most recent call last):
  File "/usr/local/bin/hpilo_cli", line 319, in <module>
    main()
  File "/usr/local/bin/hpilo_cli", line 206, in main
    results = [getattr(ilo, method)(**params)]
  File "/Library/Python/2.7/site-packages/hpilo.py", line 1038, in get_fw_version
    return self._info_tag('RIB_INFO', 'GET_FW_VERSION')
  File "/Library/Python/2.7/site-packages/hpilo.py", line 642, in _info_tag
    header, message = self._request(root)
  File "/Library/Python/2.7/site-packages/hpilo.py", line 201, in _request
    self._detect_protocol()
  File "/Library/Python/2.7/site-packages/hpilo.py", line 241, in _detect_protocol
    header, data = self._communicate(b('<RIBCL VERSION="2.0"></RIBCL>'), ILO_HTTP)
  File "/Library/Python/2.7/site-packages/hpilo.py", line 435, in _communicate
    sock.shutdown(socket.SHUT_RDWR)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ssl.py", line 292, in shutdown
    socket.shutdown(self, how)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
socket.error: [Errno 57] Socket is not connected
```

After lots of debugging I'm wondering if it SSL related and if there is a known workaround.

**Comment by seveas** [comment]
I'm not an OSX user, and have not heard of any problem like this before. Try commenting out the shutdown line.

**Comment by rexroof** [comment]
wow.  that worked.

any clue why?

**Comment by seveas** [comment]
Just a hunch. I'll do some proper investigation and will release a fix. Thanks for reporting!

**Comment by lotia** [comment]
This similar issue may provide further explanation https://code.google.com/p/py-amqplib/issues/detail?id=45 it may be due to the socket already being shutdown as is described in the issue writeup.

**Comment by seveas** [comment]
Thanks for the link, that explains. I saw your PR as well, but don't quite agree with the approach. I think I'll go for checking errno and ignore the exception if errno == ENOTCONN.

**Comment by lotia** [comment]
Thanks for the quick response. Do you want me to patch or will you handle?

**Comment by seveas** [comment]
Either is fine by me :)

**Comment by lotia** [comment]
Changed the PR as suggested.

**Comment by seveas** [comment]
Fixed in d9069ec, thanks @lotia!
